component/pyroscope/scrape/scrape.go

```diff
@@ -97,12 +97,21 @@ type ProfilingConfig struct {
 	Custom            []CustomProfilingTarget `river:"profile.custom,block,optional"`
 
 	PprofPrefix string `river:"path_prefix,attr,optional"`
+
+	allTargets map[string]ProfilingTarget
```

Comment:
Can we cache the targets elsewhere? IMO it's an anti-pattern for a config field to contain state, and it'll eventually collide with our long-term plans to serialize these objects (see #5225).

Reply:
Moved the new benchmark is still pretty good:

```
benchstat before.txt after.txt
name     old time/op    new time/op    delta
Sync-16     796µs ±13%     182µs ± 4%  -77.13%  (p=0.008 n=5+5)

name     old alloc/op   new alloc/op   delta
Sync-16     609kB ± 0%     171kB ± 0%  -71.90%  (p=0.008 n=5+5)

name     old allocs/op  new allocs/op  delta
Sync-16     8.26k ± 0%     1.55k ± 0%  -81.19%  (p=0.008 n=5+5)
```